CMakeLists.txt

```diff
@@ -242,7 +242,7 @@ add_external_subdirectory(nanopb)
 
 target_compile_definitions(
   protobuf-nanopb-static PUBLIC
-  -DPB_FIELD_32BIT -DPB_ENABLE_MALLOC
+  -DPB_FIELD_32BIT -DPB_ENABLE_MALLOC -DPB_NO_PACKED_STRUCTS=1
```

Comment:
Do you know if this setting is enabled by default on older Mac OS versions? There might be some memory usage implications if this change enables it for the first time.

Reply:
I do not know if this setting is enabled by default on older Mac OS versions.

My guess is that the unaligned pointers were always there but for whatever reason the build toolchain in macOS Monterey fails instead of warning about this.

It is true that disabling packed structs will likely cause the size of the protos in memory to increase; however, we already disable packed structs everywhere else and it hasn't been an issue that I am aware of. I can only guess that this location here was overlooked by accident.

The C++ SDK builds with `-DPB_NO_PACKED_STRUCTS=1` so we've probably just gotten incredibly lucky that the compiler flag mismatch hasn't caused any noticeable crashes or other undefined behavior.

Reply:
See https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseFirestore.podspec#L118 and https://github.com/firebase/firebase-ios-sdk/blob/master/Package.swift#L612

Reply:
@paulb777 Exactly. This is the ONLY place where `PB_NO_PACKED_STRUCTS=1` is missing. It _really_ looks like an oversight.